src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -136,486 +121,350 @@ public bool AllowAutoRedirect
         public const bool SupportsRedirectConfiguration = true;
 
         private Dictionary<string, object?>? _properties;
+        // TODO, should this be passed to the browser ?
         public IDictionary<string, object?> Properties => _properties ??= new Dictionary<string, object?>();
 
         protected internal override HttpResponseMessage Send(HttpRequestMessage request, CancellationToken cancellationToken)
         {
             throw new PlatformNotSupportedException();
         }
 
-        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage request, CancellationToken cancellationToken, bool? allowAutoRedirect)
         {
-            ArgumentNullException.ThrowIfNull(request);
-            return Impl(request, cancellationToken);
+            List<string> headerNames = new List<string>();
```

Comment:
consider pre-sizing these lists since we know the number of request headers

i don't think the overall performance of this method is a concern, but this new approach looks like it may allocate much more than the previous one did so it's possible we will get complaints about that later. (it's fine as-is)

Reply:
Yes SetObjectProperty was definitely worse, so this is a good change overall